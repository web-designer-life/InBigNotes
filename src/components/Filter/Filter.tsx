import React, { Component } from 'react';
import { TEXTS } from '../../constants';
import {
    FilterContainer,
    FilterTitle,
    FilterList,
    FilterListItem,
} from './style';

interface Props {
    filter: string,
    fetchNotes(filter: string): Function,
    filterAction(filter: string): Function,
};

interface State {
    filter: string,
};

export default class Filter extends Component<Props, State> {
    constructor(props: Props) {
        super(props);
        
        this.state = {
            filter: this.props.filter,
        };

        this.handleChangeFilter = this.handleChangeFilter.bind(this);
    };

    handleChangeFilter(evt: { target: { value: string; }; }) {
        const { value } = evt.target;

        this.props.filterAction(value);
        this.props.fetchNotes(value);
    };
      
    render() {
        return (
            <FilterContainer>
                <FilterTitle>{TEXTS.FILTER.TITLE}</FilterTitle>
                <FilterList
                    onChange={this.handleChangeFilter}
                    value={this.props.filter}
                >                 
                    <FilterListItem value={TEXTS.FILTER.NAME.VALUE}>{TEXTS.FILTER.NAME.TEXT}</FilterListItem>
                    <FilterListItem value={TEXTS.FILTER.DATE_CREATED.VALUE}>{TEXTS.FILTER.DATE_CREATED.TEXT}</FilterListItem>
                    <FilterListItem value={TEXTS.FILTER.DATE_MODIFIED.VALUE}>{TEXTS.FILTER.DATE_MODIFIED.TEXT}</FilterListItem>
                </FilterList>
            </FilterContainer>
        )
    }
};
