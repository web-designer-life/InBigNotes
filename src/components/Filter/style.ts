import styled from 'styled-components';

const FilterContainer = styled.div`
    display: flex;
    align-items: center;
`;

const FilterTitle = styled.h2`
    font-family: 'Lora', 'Arial', sans-serif;
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    line-height: 46px;
    margin: 0 30px 0 0;
    color: #5A5A5A;
`;

const FilterList = styled.select`
    font-family: 'Lora', 'Arial', sans-serif;
    font-style: normal;
    font-weight: bold;
    font-size: 30px;
    line-height: 46px;
    color: #5A5A5A;
    background-color: #FFFFFF;
    border: 5px solid #5A5A5A;
    border-radius: 20px;
    padding: 0 30px 0 30px;
    height: 80px;
    cursor: pointer;
`;

const FilterListItem = styled.option`

`;

export {
    FilterContainer,
    FilterTitle,
    FilterList,
    FilterListItem,
};
