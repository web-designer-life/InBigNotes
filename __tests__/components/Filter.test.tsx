import React from 'react';
import { shallow } from 'enzyme';
import Filter from '../../src/components/Filter';

const filterProps = {
    filterType: 'title',
    fetchNotes: (filterType: string) => Function,
    filterAction: (filterType: string) => Function,
};

describe('Filter component', () => {
    it('should render Filter component', () => {
        const component = shallow(<Filter {...filterProps} />);
        expect(component).toMatchSnapshot();
    });
});
