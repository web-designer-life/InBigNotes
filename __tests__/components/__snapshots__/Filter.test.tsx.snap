// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render Filter component 1`] = `
<styled.div>
  <styled.h2>
    Sort by:
  </styled.h2>
  <styled.select
    onChange={[Function]}
    value="title"
  >
    <styled.option
      value="title"
    >
      Name
    </styled.option>
    <styled.option
      value="created_at"
    >
      Date Created
    </styled.option>
    <styled.option
      value="updated_at"
    >
      Date Modified
    </styled.option>
  </styled.select>
</styled.div>
`;
